It is just a PEP8 error:

```
./product_variant_default_code/__openerp__.py:33:75: W291 trailing whitespace
```

There is a space at the end of the line

--- a/product_variant_default_code/README.rst
+++ b/product_variant_default_code/README.rst
@@ -1,0 +1,16 @@
+Product Variant Default Code(product_variant_default_code)
+-----------------------------------------------------------
+
+#. In 'product.template' object 'variant_reference mask' field is added
+
+#. In 'product.attribute.value' object is added the new field
+        'Attribute Code'.
+
+#. Reference mask is automatically created according to the attribute 
+        line settings on the product template. The mask can be changed
+        adaptively later on and the default code for vaiants will be
+        generated accodingly.
+
+#. Reference code field of product is calculated automatically, taking as
+        the value of the new field 'Attribute Code'.
+

--- a/product_variant_default_code/__openerp__.py
+++ b/product_variant_default_code/__openerp__.py
@@ -18,7 +18,7 @@
 {
     "name": "Product Variant Default Code",
     "version": "2.0",
-    "author": "Shine IT(http:www.openerp.cn), OdooMRP team",
+    "author": "Shine IT(http://www.openerp.cn), OdooMRP team",
     "contributors": "Tony Gu<tony@openerp.cn>",
     "category": "Product",
     "website": "http://www.odoomrp.com",

--- a/product_variant_default_code/__openerp__.py
+++ b/product_variant_default_code/__openerp__.py
@@ -30,7 +30,7 @@
     2.- In 'product.attribute.value' object is added the new field
         'Attribute Code'.
 
-    3.- Reference mask is automatically created according to the attribute 
+    3.- Reference mask is automatically created according to the attribute
         line settings on the product template. The mask can be changed
         adaptively later on and the default code for vaiants will be
         generated accodingly.

--- a/product_variant_default_code/models/product.py
+++ b/product_variant_default_code/models/product.py
@@ -109,7 +109,8 @@ class ProductProduct(models.Model):
     @api.model
     def create(self, values):
         product = super(ProductProduct, self).create(values)
-        render_default_code(product, product.reference_mask)
+        if product.reference_mask:
+            render_default_code(product, product.reference_mask)
         return product
 
 

--- a/product_variant_default_code/models/product.py
+++ b/product_variant_default_code/models/product.py
@@ -59,7 +59,7 @@ def render_default_code(product, mask):
     missing_attrs = all_attrs - set(product_attrs.keys())
     missing = dict.fromkeys(missing_attrs, PLACE_HOLDER_4_MISSING_VALUE)
     product_attrs.update(missing)
-    default_code = reference_mask.substitute(product_attrs)
+    default_code = reference_mask.safe_substitute(product_attrs)
     product.default_code = default_code
 
 

--- a/product_variant_default_code/models/product.py
+++ b/product_variant_default_code/models/product.py
@@ -15,9 +15,8 @@
 #    along with this program.  If not, see http://www.gnu.org/licenses/.
 #
 ##############################################################################
-from openerp import models, fields, api
+from openerp import models, fields, api, _
 from openerp.exceptions import except_orm
-from openerp.tools.translate import _
 import re
 from string import Template
 from collections import defaultdict

